### PR TITLE
Add a modern linux-x86 config target

### PR DESCRIPTION
**Pull request by levitte**
##### Checklist

- [x] CLA is signed

##### Description of change

'linux-x86' is similar to 'linux-x86_64' but uses -m32 rather than -m64.

'linux-elf' is kept as a legacy target for Linux machines where the compiler doesn't understand -m32.

`config` is updated to check what the compiler understands and choose 'linux-x86' or 'linux-elf' accordingly.

### CODE REVIEW

**Comment by levitte** [comment]
I have added 1.1.0, as I view this as a bug correction.  On x86_64 multi-arch Linux machines, 'linux-elf' doesn't build.

File: Configurations/10-main.conf [hunk 1]
```diff
@@ -748,6 +750,21 @@ sub vms_info {
         perlasm_scheme   => "a.out",
     },
 
+    #### X86 / X86_64 targets
+    "linux-x86" => {
+        inherit_from     => [ "linux-generic32", asm("x86_asm") ],
+        cflags           => add(picker(default => "-m32 -DL_ENDIAN",
+                                       release => "-fomit-frame-pointer")),
+        bn_ops           => "BN_LLONG",
+        perlasm_scheme   => "elf",
```
Comment (dot-asm) [inline Configurations/10-main.conf]: perlasm_scheme is defined in x86_asm, no need to redefine it here.

Comment (levitte) [inline Configurations/10-main.conf]: Will remove

Comment (levitte) [inline Configurations/10-main.conf]: And uhmm no.  `perlasm_scheme` is defined in x86_elf_asm, not in `x86_asm`

File: Configurations/10-main.conf [hunk 2]
```diff
@@ -731,6 +731,8 @@ sub vms_info {
     },
 
     #### IA-32 targets...
+    #### These targets are a bit aged and are to be used on older Linux
```
Comment (levitte) [inline Configurations/10-main.conf]: Actually, I think it's fitting to add "two" either way

File: Configurations/10-main.conf (hunk 1 above)
Comment (dot-asm) [inline Configurations/10-main.conf]: Right. I was assuming that you would inherit x86_elf_asm, as linux-elf does. Well, personally I'd prefer to see x86_elf_asm, but it doesn't matter much...

**Comment by dot-asm** [comment]
+1 (applies even if you choose to replace x86_asm with x86_elf_asm and omit perlasm_scheme).

**Comment by levitte** [comment]
Merged

**Pull request closed without merging**